CRM/Core/BAO/MessageTemplate.php

```diff
@@ -87,10 +87,12 @@ public static function add(&$params) {
           }
         }
         else {
-          if (!empty($params['workflow_id']) && !CRM_Core_Permission::check('edit system workflow message templates')) {
-            throw new \Civi\API\Exception\UnauthorizedException(ts('%1', [1 => $systemWorkflowPermissionDeniedMessage]));
+          if (!empty($params['workflow_id'])) {
+            if (!CRM_Core_Permission::check('edit system workflow message templates')) {
+              throw new \Civi\API\Exception\UnauthorizedException(ts('%1', [1 => $systemWorkflowPermissionDeniedMessage]));
+            }
           }
-          if (!CRM_Core_Permission::check('edit user-driven message templates')) {
+          elseif (!CRM_Core_Permission::check('edit user-driven message templates')) {
```

Comment:
Note that this needs to be elseif so that we don't run the check if workflow_id is set